zcash_client_sqlite/src/wallet/common.rs

```diff
@@ -321,11 +319,7 @@ where
            OR stx.expiry_height IS NULL -- the spending tx will not expire
            OR stx.expiry_height >= :target_height -- the spending tx is unexpired
          )
-         GROUP BY
-            rn.id, t.txid, rn.{output_index_col},
-            rn.diversifier, rn.value, {note_reconstruction_cols}, rn.commitment_tree_position,
-            ufvk, rn.recipient_key_scope, 
-            t.block, scan_state.max_priority"
+         GROUP BY rn.id"
```

Comment:
I added checks in the test that would catch a bug here if one existed, but it didn't because we were already grouping by `rn.id` as it's a returned column.